chain/get_ancestors.go

```diff
@@ -85,57 +88,35 @@ func GetRecentAncestors(ctx context.Context, base types.TipSet, chainReader Read
 // CollectTipSetsOfHeightAtLeast collects all tipsets with a height greater
 // than or equal to minHeight from the input channel.  Precondition, the input
 // channel contains interfaces which may be tipsets or errors.
-func CollectTipSetsOfHeightAtLeast(ctx context.Context, ch <-chan interface{}, minHeight *types.BlockHeight) ([]types.TipSet, error) {
-	var ret []types.TipSet
+func CollectTipSetsOfHeightAtLeast(ctx context.Context, chainReader ReadStore, ts *types.TipSet, minHeight *types.BlockHeight) (ret []types.TipSet, err error) {
```

Comment:
question from curiosity: why is delegating all ctx error passing to `ts.GetNext` (the only method waiting on the context) a bad thing?  Is it wrong to assume that this call will handle context errors correctly?  It seems like you are advocating for assuming that other parts of the system are doing things wrong which makes each individual function more robust to programmer error elsewhere in the system but potentially adds in unnecessary redundancy.

Reply:
Since this is going to use the new TipSet iterator in `chain/traversal.go`, I'll just change said iterator to respect `ctx.Err()` on `Next()`. That should cover this case, the next comment, and any future uses of `IterAncestors`.